src/renderer/qssWidget/js/qssOfficeWidget.js

```diff
@@ -0,0 +1,391 @@
+                includeSource: "settingAlter"
+            }
+        },
+        selectors: {
+            heightListenerContainer: ".flc-qssOfficeWidget-controls",
+            menuControlsWrapper: ".flc-qssOfficeWidget-controlsWrapper",
+            menuControls: ".flc-qssOfficeWidget-controls",
+            footerTip: ".flc-qssMenuWidget-footerTip"
+        },
+        enableRichText: true,
+        activationParams: {},
+        closeDelay: 1200,
+        components: {
+            repeater: {
+                type: "gpii.psp.repeater",
+                container: "{office}.dom.menuControls",
+                options: {
+                    model: {
+                        disabled: "{office}.model.disabled",
+                        value: "{office}.model.setting.value",
+                        styles: "{office}.model.setting.styles"
+                    },
+                    modelRelay: {
+                        items: {
+                            target: "items",
+                            singleTransform: {
+                                type: "fluid.transforms.free",
+                                func: "gpii.qssWidget.office.getRepeaterItems",
+                                args: [
+                                    "{office}.model.setting"
+                                ]
+                            }
+                        }
+                    },
+                    dynamicContainerMarkup: {
+                        container: "<div role=\"radio\" class=\"%containerClass fl-qssWidgetMenu-item fl-focusable\" tabindex=\"0\"></div>",
+                        containerClassPrefix: "flc-qssWidgetMenu-item"
+                    },
+                    handlerType: "gpii.qssWidget.office.presenter",
+                    markup: null,
+                    styles: {
+                        disabled: "disabled"
+                    },
+                    events: {
+                        onItemFocus: null
+                    },
+                    listeners: {
+                        "onCreate.enable": {
+                            this: "{that}.container",
+                            method: "removeClass",
+                            args: ["{that}.options.styles.disabled"]
+                        }
+                    }
+                }
+            },
+            closeTimer: {
+                type: "gpii.app.timer",
+                options: {
+                    listeners: {
+                        "onTimerFinished.closeWidget": {
+                            func: "{qssWidget}.close",
+                            args: ["{office}.keyboardEvent"]
+                        }
+                    }
+                }
+            }
+        },
+        listeners: {
+            "onCreate.visibilityChange": {
+                funcName: "gpii.qssWidget.office.addVisibilityChangeListener",
+                args: ["{closeTimer}"]
+            },
+            "onDestroy.visibilitychange": {
```

Comment:
Capitalise namespace for consistency to match the onCreate listener - but in any case all these listeners need to be factored up

Reply:
Done.